### PR TITLE
Implement CLI for the launcher

### PR DESCRIPTION
**Pull request by HirotoShioi**
https://github.com/input-output-hk/cardano-shell/issues/271

TODO: Merge https://github.com/input-output-hk/cardano-shell/pull/255 and rebase from master

How to run: `:main -c /path/to/config.yaml`

### CODE REVIEW

File: cardano-launcher/src/Cardano/Shell/CLI.hs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+newtype LauncherOptionPath = LauncherOptionPath
+    { getLauncherOptionPath  :: FilePath
+    }
+
+-- | Default path to the launcher-config.yaml file
+--
+-- This will be used if the filepath was not provided via CLI
+getDefaultConfigPath :: IO FilePath
+getDefaultConfigPath = do
+    launcherDir <- takeDirectory <$> getExecutablePath
```
Comment (ksaric) [inline cardano-launcher/src/Cardano/Shell/CLI.hs]: Are you sure about this one? Did you test this somehow? I presume that the executable will be provided once we create an executable out of it, but I'm still not sure which one.
```
$> stack ghci

> import           System.Environment (getExecutablePath)
> getExecutablePath
"/home/ksaric/.stack/programs/x86_64-linux/ghc-8.6.5/lib/ghc-8.6.5/bin/ghc"
```

Comment (HirotoShioi) [reply cardano-launcher/src/Cardano/Shell/CLI.hs]: This is used for fetching configuration file on both Mac and Windows.

Comment (ksaric) [reply cardano-launcher/src/Cardano/Shell/CLI.hs]: I understand that, what I'm asking is what path will be returned here if we run it using Nix, for example, and would that work?

Comment (cleverca22) [reply cardano-launcher/src/Cardano/Shell/CLI.hs]: that is why linux uses the `--config` argument on the command line, rather then expecting the cfg to be in the same dir as the binary

https://github.com/input-output-hk/daedalus/blob/develop/installers/nix/linux.nix#L47-L48

**Pull request merged**